fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -132,6 +132,30 @@ private void processAddColumn(AddColumnClause alterClause, OlapTable olapTable,
                           indexSchemaMap, newColNameSet);
     }
 
+    private void processAddColumn(AddColumnClause alterClause, Table externalTable) throws DdlException {
+        Column column = alterClause.getColumn();
+        ColumnPosition columnPos = alterClause.getColPos();
+        Set<String> newColNameSet = Sets.newHashSet(column.getName());
+
+        List<Column> modIndexSchema = externalTable.getBaseSchema();
+        addColumnInternal(externalTable, column, columnPos, modIndexSchema, newColNameSet);
+        externalTable.setNewFullSchema(modIndexSchema);
```

Comment:
Should not set new schema to `externalTable` here directly.
The other alter clause may throw exception and the entire alter operation will fail, and you have to rollback all changes to the table.

A correct way to to modify a "copied" schema of external table, and set the new schema at the end of all alter clauses.

Reply:
yes， it is a problem。I will set the new schema at the end of alter external table.